src/js/net.js

```diff
@@ -279,7 +279,11 @@ function connect(socket, ip, port) {
     }
   };
 
-  socket._handle.connect(ip, port, afterConnect);
+  var err = socket._handle.connect(ip, port, afterConnect);
+  if (err) {
+    emitError(socket, new Error('connect failed - status: ' +
+      TCP.errname(err)));
+  }
```

Comment:
@chokobole ``EmitError`` is a function defined locally in the same file and used only twice. I've patched this function to call ``process.nextTick`` (same code as your suggestion, but in a function). Both uses of the function are consistent with a deferred event propagation.